packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,219 @@
+  ///
+  /// [borderRadius] sets the circular border radius for the [Column]
+  /// encapsulating [children] rows. Defaults to 10.0 for the standard
+  /// edge-to-edge style.
+  ///
+  /// [decoration] sets the decoration for the section. Defaults to a
+  /// [CupertinoColors.secondarySystemBackground] background color.
+  const CupertinoFormSection.centered({
+    this.title,
+    this.titlePadding =
+        const EdgeInsetsDirectional.fromSTEB(16.5, 16.0, 16.5, 10.0),
+    this.rowsPadding =
+        const EdgeInsetsDirectional.fromSTEB(16.5, 0.0, 16.5, 10.0),
+    this.borderRadius = 10.0,
+    required this.children,
+    this.decoration =
+        const BoxDecoration(color: CupertinoColors.secondarySystemBackground),
+  })  : _type = _CupertinoFormSectionType.centered,
+        super();
+
+  final _CupertinoFormSectionType _type;
+
+  /// Sets the form section title.
+  final String? title;
+
+  /// Sets padding for the section title.
+  ///
+  /// Defaults to the standard iOS padding, which is determined
+  /// using https://github.com/flutter/platform_tests/tree/master/ios_widget_catalog_compare.
+  final EdgeInsetsGeometry titlePadding;
+
+  /// Padding for the the [Column] encapsulating the [children].
+  ///
+  /// Defaults to zero padding if constructed with standard
+  /// [CupertinoFormSection] constructor. Defaults to the standard notched-style
+  /// iOS padding when constructing with [CupertinoFormSection.centered].
+  /// Determined using https://github.com/flutter/platform_tests/tree/master/ios_widget_catalog_compare.
+  final EdgeInsetsGeometry rowsPadding;
+
+  /// The list of rows in the section.
+  final List<Widget> children;
+
+  /// Sets the decoration for the section.
+  ///
+  /// Defaults to a [CupertinoColors.secondarySystemBackground] background
+  /// color.
+  final BoxDecoration decoration;
+
+  /// Sets the circular border radius for the [Column] encapsulating [children]
+  /// rows.
+  ///
+  /// Defaults to 0.0 when constructed with standard [CupertinoFormSection]
+  /// constructor. Defaults to 10.0 when constructing with
+  /// [CupertinoFormSection.centered].
+  final double borderRadius;
+
+  @override
+  State<StatefulWidget> createState() => _CupertinoFormSectionState();
+}
+
+class _CupertinoFormSectionState extends State<CupertinoFormSection> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const Divider longDivider = Divider(
+      endIndent: 0,
+      indent: 0,
+      color: CupertinoColors.separator,
+      height: 0.0,
+    );
+
+    const Divider shortDivider = Divider(
+      endIndent: 0,
+      indent: 15.0,
+      color: CupertinoColors.separator,
+      height: 0.0,
+    );
+
+    final List<Widget> children =
+        (widget._type == _CupertinoFormSectionType.base)
+            ? <Widget>[longDivider]
+            : <Widget>[];
+
+    widget.children
+        .sublist(0, widget.children.length - 1)
```

Comment:
add another sanity assertion that it's not null here. Generally, feel free to liberally sprinkle asserts to check your assumptions